src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -147,21 +150,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   }
 
   /**
-   * Returns the Entity in database that has {@code url}, or
+   * Returns the Entity in database that has {@code videoId}, or
    * {@code Optional.empty()} if one doesn't exist.
    */
-  // TODO: Use a filter to avoid fetching all lectures.  See: #185.
-  private Optional<Entity> checkUrlInDatabase(String url) {
-    Query query = new Query(LectureUtil.KIND);
+  private Optional<Entity> queryForLectureWithVideoId(String videoId) {
```

Comment:
Would a better method name be 'lectureWithVideoId'. "queryFor" seems to explain how you obtain the lecture, so I am not sure it is necessary.

Reply:
It doesn't always find/return a lecture with a matching video id.  The original name was `check...`, but we felt like that implied a boolean return value, and changed it to this.  If you think `lectureWithVideoId` is better i can change it. :)

Reply:
Optional<Entity> implies it might not return a lecture with video id. This is a getter method and typically getter methods are called getFoo() or foo(). 

This is a nit mostly, so you can keep it as is if you prefer your name.

Reply:
This made more sense to us, so i'll keep it